app/controllers/registrations_controller.rb

```diff
@@ -35,22 +35,22 @@ def start
       email: params.dig(:user, :email),
       password: params.dig(:user, :password),
       password_confirmation: params.dig(:user, :password),
-      phone: params.dig(:user, :phone),
-      enforce_has_mfa: true,
+      phone: params.dig(:user, :phone) if request_auth_wants_mfa?,
```

Comment:
This syntax doesn't work in this context, I think you'll have to do `request_auth_wants_mfa? ? params.dig(:user, :phone) : nil` (or extract that to a variable since you'll do the same thing on line 51)